network/src/peer_manager/mod.rs

```diff
@@ -585,7 +590,7 @@ where
             for handler in self.connection_event_handlers.iter_mut() {
                 handler
                     .push(
-                        (peer_id, ProtocolId::default()),
+                        (peer_id, ProtocolId::from_static(b"NewPeer")),
```

Comment:
spoke offline about this. doesn't look like it's going to be sufficient, so @bothra90 is going to add a separate channel specifically for NewPeer/LostPeer events -> client handlers.